src/commonMain/kotlin/com/algolia/search/transport/internal/Transport.kt

```diff
@@ -82,29 +79,45 @@ internal class Transport(
         body: String? = null,
     ): T {
         val hosts = callableHosts(callType)
-        val requestBuilder = httpRequestBuilder(httpMethod, path, requestOptions, body, callType)
         val errors by lazy(LazyThreadSafetyMode.NONE) { mutableListOf<Throwable>() }
+        val requestBuilder = httpRequestBuilder(httpMethod, path, requestOptions, body)
 
         for (host in hosts) {
             requestBuilder.url.host = host.url
             try {
+                setTimeout(requestBuilder, requestOptions, callType, host)
                 return httpClient.request<T>(requestBuilder).apply {
```

Comment:
Not sure how it works. Will `apply { mutex.withLock { host.reset() } }` only be called in case of successful response?

Reply:
Yes exactly!

Reply:
Great, the rest LGTM 👍